dex/src/main/java/io/crate/data/RowsBatchIterator.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.data;
+
+import io.crate.concurrent.CompletableFutures;
+
+import java.util.Iterator;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * BatchIterator implementation that is backed by {@link Iterable<Row>}.
+ */
+public class RowsBatchIterator implements BatchIterator {
+
+    private final Iterable<Row> rows;
+    private Iterator<Row> it;
+    private Row currentRow = BatchIterator.OFF_ROW;
+
+    public RowsBatchIterator(Iterable<Row> rows) {
+        this.rows = rows;
+        this.it = rows.iterator();
+    }
+
+    @Override
+    public void moveToStart() {
+        it = rows.iterator();
+        currentRow = OFF_ROW;
+    }
+
+    @Override
+    public boolean moveNext() {
+        if (it.hasNext()) {
+            currentRow = it.next();
+            return true;
+        }
+        currentRow = OFF_ROW;
+        return false;
+    }
+
+    @Override
+    public void close() {
+    }
```

Comment:
I'd like to not do that because this implementation is currently not taking care of close at all.
Once I null it here I'd also have to add null checks in all other methods.

I wanted to avoid that to have a "zero overhead" implementation as a baseline for performance tests (See the comment in the jmh benchmark)